app/services/claim_change_history/claim_history_service.rb

```diff
@@ -235,6 +240,40 @@ def process_decision_issue_and_task_events(change_data)
     end
   end
 
+  def process_request_issue_modification_events(change_data)
+    issue_modification_request_id = change_data["issue_modification_request_id"]
+
+    return unless issue_modification_request_id
+
+    is_processed = @processed_issue_modification_request_ids.include?(issue_modification_request_id)
+
+    if issue_modification_request_id && !is_processed
+      @processed_issue_modification_request_ids.add(issue_modification_request_id)
+      save_events(ClaimHistoryEvent.create_issue_modification_request_event(change_data))
+      save_events(ClaimHistoryEvent.create_edited_request_issue_events(change_data))
+    end
+    # if imr version doesn't have status it means, imr has been made but no action yet taken from admin
+    process_pending_status_event(change_data) if !is_processed
+  end
+
+  # processed_issue_modification_task_id stores all the task id that has already
+  # been processed so that it prevents the duplicate entry of pending event based on change_ data.
```

Comment:
I would move this down into the method where the processed_issue_modification_task_id is being used.
```suggestion
  # been processed so that it prevents the duplicate entry of pending event based on change_data.
```